src/main/java/com/allenai/ml/linalg/SparseVector.java

```diff
@@ -87,5 +89,5 @@ public double dotProduct(Vector other) {
           result += val * other.at(idx);
```

Comment:
Ah, yes you're right I've written code like the above before. I did it access to an iterator (like `Vector.Iterator`) rather than the direct arrays; I think you can still do the above efficiently, since you're only relying on advancing each of the cursors and retriving the index and value.